### PR TITLE
Lock down site for beta

### LINKED ISSUE #597: Shut down for alpha

### PR DESCRIPTION
**Pull request by ritmatter**
Fixes #597

### CODE REVIEW

File: api/routes/AuthRoutes.js [hunk 1]
```diff
@@ -11,15 +11,26 @@ module.exports = function(router, passport) {
   as loggedin or not.
   */
 
-  router.post('/login/facebook',
+  router.post('/login/facebook', acl.isAdmin,
```
Comment (ritmatter) [inline api/routes/AuthRoutes.js]: Temporarily require admin to login

File: api/routes/AuthRoutes.js [hunk 2]
```diff
@@ -11,15 +11,26 @@ module.exports = function(router, passport) {
...
     passport.authenticate('facebook-token'), function (req, res) {
       var user = req.user || {};
       return res.status(200).send(user);
     });
 
+  // TODO: Revert this for the beta release
   router.route('/loggedin')
     .get(function(req, res) {
-      return res.status(200).send(req.isAuthenticated());
+      if (process.env.NODE_ENV !== 'production') {
```
Comment (ritmatter) [inline api/routes/AuthRoutes.js]: Ignore lockdown in dev mode

File: api/routes/AuthRoutes.js [hunk 3]
```diff
@@ -11,15 +11,26 @@ module.exports = function(router, passport) {
...
+        return res.status(200).send(req.isAuthenticated());
+      }
+      if (!req.user) {
```
Comment (ritmatter) [inline api/routes/AuthRoutes.js]: In production, only let our facebook accounts through

File: api/routes/AuthRoutes.js [hunk 4]
```diff
@@ -29,7 +40,7 @@ module.exports = function(router, passport) {
     });
 
   router.route('/login')
-    .post(function(req, res, next) {
+    .post(acl.isAdmin, function(req, res, next) {
```
Comment (ritmatter) [inline api/routes/AuthRoutes.js]: temporarily require admin to login

File: config/acl.js [hunk 5]
```diff
@@ -5,6 +5,21 @@ var whitelist = require('./whitelist.js');
 var acl = {
   // Check if the user has admin privileges
   isAdmin: function(req, res, next) {
+
+    // TODO: remove this after Beta
+    // No need to enforce admin on a dev instance
+    if (process.env.NODE_ENV !== 'production') {
```
Comment (ritmatter) [inline config/acl.js]: avoid lockdown in dev mode

File: config/acl.js [hunk 6]
```diff
@@ -5,6 +5,21 @@ var whitelist = require('./whitelist.js');
...
+      next();
+      return;
+    }
+
+    // TODO: remove this after Beta release
+    // This hack will allow us to login with our facebook accounts
+    if (req.body.access_token) {
```
Comment (ritmatter) [inline config/acl.js]: let facebook accounts login anyway

File: config/pass.js [hunk 7]
```diff
@@ -46,6 +46,12 @@ module.exports = function(passport, LocalStrategy, FacebookTokenStrategy) {
           return done(err);
         }
 
+        // TODO: remove this once we are in Beta
+        // This hack allows us to login with facebook
```
Comment (ritmatter) [inline config/pass.js]: do not let other facebook users login

File: js/components/Login.jsx [hunk 8]
```diff
@@ -12,7 +12,8 @@ var Login = React.createClass({
   getInitialState: function() {
     return {
       error: false,
-      forgotPassword: false
+      forgotPassword: false,
```
Comment (ritmatter) [inline js/components/Login.jsx]: This was a separate glitch we were having

**Comment by krazemon** [comment]
LGTM

**Pull request merged**